scripts/scil_score_tractogram.py

```diff
@@ -0,0 +1,544 @@
+            affine = gt_img.affine
+            dimensions = gt_mask.shape
+        else:
+            gt_sft = load_tractogram_with_reference(parser, args, gt_bundle)
+            gt_sft.remove_invalid_streamlines()
+            gt_sft.to_vox()
+            gt_sft.to_corner()
+            affine, dimensions, _, _ = gt_sft.space_attributes
+            gt_mask = compute_tract_counts_map(gt_sft.streamlines,
+                                               dimensions).astype(np.int16)
+        gt_inv_mask = np.zeros(dimensions, dtype=np.int16)
+        gt_inv_mask[gt_mask == 0] = 1
+        gt_mask[gt_mask > 0] = 1
+        gt_bundle_masks.append(gt_mask)
+        gt_bundle_inv_masks.append(gt_inv_mask)
+
+    # If endpoints without heads/tails are loaded, split them and continue
+    # normally after. Q/C of the output is important
+    if args.gt_endpoints:
+        tails = []
+        heads = []
+        for mask_filename in args.gt_endpoints:
+            mask = nib.load(mask_filename).get_fdata().astype(np.int16)
+            head, tail = split_heads_tails_kmeans(mask)
+
+            basename = os.path.basename(
+                split_name_with_nii(mask_filename)[0])
+            tail_filename = os.path.join(
+                args.out_dir, '{}_tail.nii.gz'.format(basename))
+            head_filename = os.path.join(
+                args.out_dir, '{}_head.nii.gz'.format(basename))
+            nib.save(nib.Nifti1Image(head, affine), head_filename)
+            nib.save(nib.Nifti1Image(tail, affine), tail_filename)
+
+            tails.append(tail_filename)
+            heads.append(head_filename)
+
+        args.gt_tails, args.gt_heads = tails, heads
+        args.gt_endpoints = None
+
+    # Load the endpoints heads/tails, keep the correct combinations
+    # separately from all the possible combinations
+    tc_data = []
+    all_data = []
+    fused_masks = np.zeros(dimensions)
+    tc_filenames = list(zip(args.gt_tails, args.gt_heads))
+
+    indices = []
+    for mask_1_filename, mask_2_filename in tc_filenames:
+        mask_1 = nib.load(mask_1_filename).get_fdata().astype(np.int16)
+        mask_2 = nib.load(mask_2_filename).get_fdata().astype(np.int16)
+        _, loops_1_ind = filter_grid_roi(sft, mask_1, 'both_ends', False)
+        _, loops_2_ind = filter_grid_roi(sft, mask_2, 'both_ends', False)
+        indices.extend(loops_1_ind)
+        indices.extend(loops_2_ind)
+        # if args.dilate_endpoints:
+        # That would be done here.
+
+        fused_masks += mask_1 + mask_2
+        tc_data.append((mask_1, mask_2))
+        all_data.extend([mask_1, mask_2])
+    fused_masks[fused_masks > 0] = 1
+
+    indices = np.unique(indices).tolist()
+    no_conn_streamlines = []
+    no_conn_streamlines.extend(sft.streamlines[indices])
+
+    overlapping_roi = []
+    all_rois = args.gt_heads + args.gt_tails
+    for roi1, roi2 in itertools.combinations(all_rois, 2):
+        identify_overlapping_roi(roi1, roi2, overlapping_roi)
```

Comment:
Adding on @arnaudbore's comment elsewhere, it'd be less confusing if the function had an explicit return instead of using side-effects